### PR TITLE
MRG: Standardizing bids_root / output_path for the read/write API

### PR DESCRIPTION
**Pull request by adam2392**
PR Description
--------------

Homogenizing the bids_root/output_path naming within the read/write API of mne_bids. Closes and addresses issue: #213 

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [x] All comments resolved
- [x] This is not your own PR
- [x] All CIs are happy
- [x] PR title starts with [MRG]
- [x] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/master/doc/whats_new.rst) is updated
- [x] PR description includes phrase "closes <#issue-number>"
- [x] Commit history does not contain any merge commits

### CODE REVIEW

File: mne_bids/write.py [hunk 1]
```diff
@@ -964,7 +964,7 @@ def make_dataset_description(path, name=None, data_license=None,
     _write_json(fname, description, overwrite=True, verbose=verbose)
 
 
-def write_raw_bids(raw, bids_basename, output_path, events_data=None,
+def write_raw_bids(raw, bids_basename, bids_root, events_data=None,
```
Comment (jasmainak) [inline mne_bids/write.py]: are we willing to break this without deprecation cycle @agramfort ?

Comment (sappelhoff) [reply mne_bids/write.py]: I'd be willing to accept it without a deprecation cycle, given that it's a fairly obvious change and easy to adapt to. We can also make it prominent in the changelog.

**Comment by adam2392** [comment]
Hi @sappelhoff @jasmainak I'm not sure what's going on, but I'm getting an error in the CI, but I don't see how it could be related to any changes I made.

```
        with pytest.raises(ValueError, match='All measurement dates are None'):
>           get_anonymization_daysback(raw)
E           Failed: DID NOT RAISE <class 'ValueError'>
mne_bids\tests\test_write.py:488: Failed
```

Would you be able to advise me on how to proceed? It looks like everything else is passing and works as intended.

**Comment by jasmainak** [comment]
@adam2392 I think it's related to this PR on MNE master: https://github.com/mne-tools/mne-python/pull/7104. Can you verify locally on your computer that tests pass before this PR?

Let's raise an issue and discuss what's the right fix for that.

Also you'd need to update `whats_new.rst` on this PR

File: mne_bids/tests/test_write.py [hunk 2]
```diff
@@ -717,7 +717,7 @@ def test_set(_bids_validate):
                       extra_params=dict(foo='bar'))
 
     with pytest.raises(FileExistsError, match="already exists"):  # noqa: F821
-        write_raw_bids(raw, bids_basename, output_path=output_path,
+        write_raw_bids(raw, bids_basename, bids_root=output_path,
```
Comment (jasmainak) [inline mne_bids/tests/test_write.py]:
```suggestion
        write_raw_bids(raw, bids_basename, bids_root=bids_root,
```

why do you still keep some references to `output_path` ?

File: mne_bids/write.py (hunk 1 above)
Comment (jasmainak) [reply mne_bids/write.py]: It's a good way to check if we really have users :-)

**Comment by adam2392** [comment]
@jasmainak it seems to work when that PR isn't there. However, I'm not a master at setting up an environment that can switch easily between different repo versions, so I am a bit hesitant.

Also, I've updated the relevant sections in read_raw_bids, write_raw_bids and make_bids_folder to just use the keyword `bids_root`. Lmk what you think.

File: doc/whats_new.rst [hunk 3]
```diff
@@ -22,6 +22,7 @@ Current
 Changelog
 ~~~~~~~~~
 
+- Standardized `bids_root` and `output_path` arguments in :func:`read_raw_bids`, :func:`write_raw_bids`, and :func:`make_bids_folder` to just `bids_root`, by `Adam Li`_ (`#303 <https://github.com/mne-tools/mne-bids/pull/303>`_)
```
Comment (jasmainak) [inline doc/whats_new.rst]: You need to add your name at the bottom so that it renders.

File: mne_bids/tests/test_write.py [hunk 4]
```diff
@@ -251,7 +251,7 @@ def test_fif(_bids_validate):
 
     bids_basename2 = bids_basename.replace(subject_id, subject_id2)
     raw = mne.io.read_raw_fif(raw_fname2)
-    bids_output_path = write_raw_bids(raw, bids_basename2, output_path,
+    bids_bids_root = write_raw_bids(raw, bids_basename2, bids_root,
```
Comment (agramfort) [inline mne_bids/tests/test_write.py]: Revert this one

**Comment by jasmainak** [comment]
let's merge after #307 is in to make sure nothing breaks.

**Comment by jasmainak** [comment]
@adam2392 you need to rebase.

Please set the title to MRG when you're ready

**Comment by jasmainak** [comment]
Something went wrong during rebase @adam2392 . The diff doesn't make sense

**Comment by jasmainak** [comment]
Can you try rebasing again?

**Comment by adam2392** [comment]
> Can you try rebasing again?

Sorry I don't think it is working, or I am breaking something. I tried:

`git rebase master`

And it seems to just be adding extra commits. Sorry, very rarely using rebase :p... Some advice in how to proceed would be helpful. 

I'm going off this doc: https://git-scm.com/book/en/v2/Git-Branching-Rebasing

**Comment by jasmainak** [comment]
@adam2392 can you make sure you have the latest `master` from `upstream`?

If you do "git checkout master" and then type "git log", do you see the latest commits? If not, you may want to delete your local master branch entirely using "git branch -D master" and fetch a fresh copy "git fetch upstream master:master". Then do a rebase using "git checkout bidsroot" and "git rebase master"

**Comment by jasmainak** [comment]
Also make to resolve any conflicts that you get during the rebase.

**Comment by sappelhoff** [comment]
This looks like something went wrong during rebasing :-)

what's the situation @adam2392 - do you need help?

**Comment by adam2392** [comment]
> This looks like something went wrong during rebasing :-)
> 
> what's the situation @adam2392 - do you need help?

Hey @sappelhoff I am a bit embarrassed :p because I tried figuring this out, and I for the life of me can't figure out what's wrong. Here's what I did, which I assumed to be correct:

1. git checkout master
2. git fetch upstream master
3. git checkout bidsroot
4. git rebase master
5. -> fix up conflicts
6. Pushed up changes, and then ran tests locally:

I consistently get these errors that aren't making sense anymore because I wasn't changing anything related (I believe). Was gonna take another look once I wrap up a few things, but lmk if you need more info / spot a bug in my procedure...

**Comment by jasmainak** [comment]
At this point, it might be faster to either:

Go back to the commit before the rebase -- if you made a backup branch, or through "$ git reflog". Then, squash all your commits into one and then do a rebase. If you don't squash, you have to keep resolving conflicts for each of your commits.

or alternatively, if the above sounds like Greek:

just rename this branch to something else locally and start a new branch with the name `bidsroot` that branches off from latest master, and force push that branch here.

and final easiest alternative:

make a new branch and a new PR ... :-)

May the force be with you! :-)

**Comment by jasmainak** [comment]
> Hey @sappelhoff I am a bit embarrassed :p because I tried figuring this out, and I for the life of me can't figure out what's wrong. Here's what I did, which I assumed to be correct:

>    git checkout master
>   git fetch upstream master

I suggest changing this part a bit. Instead of going to `master` branch, just delete the local `master` branch and fetch a fresh copy every time.

```sh
$ git branch -D master
$ git fetch master:master
$ git checkout bidsroot
$ git rebase master
```

**Comment by adam2392** [comment]
> At this point, it might be faster to either:
> 
> Go back to the commit before the rebase -- if you made a backup branch, or through "$ git reflog". Then, squash all your commits into one and then do a rebase. If you don't squash, you have to keep resolving conflicts for each of your commits.
> 
> or alternatively, if the above sounds like Greek:
> 
> just rename this branch to something else locally and start a new branch with the name `bidsroot` that branches off from latest master, and force push that branch here.
> 
> and final easiest alternative:
> 
> make a new branch and a new PR ... :-)
> 
> May the force be with you! :-)

The force was w/ me and I managed to fix it ^ looks right to me at least. Thanks for helping a git-newbie out :)

File: .gitignore [hunk 5]
```diff
@@ -1,4 +1,5 @@
 # Distribution / packaging
+mne-python/*
```
Comment (jasmainak) [inline .gitignore]: why this? you should not put one git folder inside another

File: mne_bids/tests/test_write.py (hunk 4 above)
Comment (jasmainak) [reply mne_bids/tests/test_write.py]: this is not yet addressed

File: doc/whats_new.rst (hunk 3 above)
Comment (jasmainak) [reply doc/whats_new.rst]: Looks like you're missing some commits. I remember you fixed this.

**Comment by jasmainak** [comment]
Merged, thanks @adam2392 !

**Pull request merged**